### PR TITLE
Efforts to reduce warnings

### PR DESCRIPTION
**Pull request by gfx**
I've found that `RUBYOPT=-w` produces lots of warnings. In fact, I found it in testing a ruby gem that depends on rouge, which output is flushed by lots of warnings from rouge. 

This PR reduces them; some are powered by rubocop. However, some warnings are difficult to solve -- please try `rake RUBYOPT=-w` -- but I'll continue to reduce them in some separate PRs.

@jneen what do you think of this kind of efforts?

### CODE REVIEW

File: lib/rouge/lexers/elixir.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Elixir < RegexLexer
         rule /#.*$/, Comment::Single
         rule %r{\b(case|cond|end|bc|lc|if|unless|try|loop|receive|fn|defmodule|
              defp?|defprotocol|defimpl|defrecord|defmacrop?|defdelegate|
-             defexception|exit|raise|throw|unless|after|rescue|catch|else)\b(?![?!])|
+             defexception|exit|raise|throw|after|rescue|catch|else)\b(?![?!])|
```
Comment (gfx) [inline lib/rouge/lexers/elixir.rb]: `unless` is here!

File: lib/rouge/lexers/mosel.rb [hunk 2]
```diff
@@ -73,7 +73,7 @@ def self.analyze_text(text)
         getbstat getdualray getiis getiissense getiistype getinfcause getinfeas getlb getloadedlinctrs getloadedmpvars getname getprimalray getprobstat getrange getsensrng getsize getsol getub getvars
         implies indicator isiisvalid isintegral loadbasis
         loadmipsol loadprob
-        maximize, minimize
```
Comment (gfx) [inline lib/rouge/lexers/mosel.rb]: Good job, rubocop!

File: Gemfile [hunk 3]
```diff
@@ -5,6 +5,8 @@ gemspec
 gem 'minitest', '~> 4.0'
 gem 'wrong'
 
+gem 'rubocop'
```
Comment (segiddins) [inline Gemfile]: I highly recommend adding a requirement to the rubocop dependency

File: lib/rouge/formatters/html.rb [hunk 4]
```diff
@@ -26,7 +26,7 @@ def safe_span(tok, safe_val)
         end
       end
 
-    private
+
```
Comment (segiddins) [inline lib/rouge/formatters/html.rb]: you can use `private_const` if you'd like

Comment (gfx) [inline lib/rouge/formatters/html.rb]: Oh, I didn't know it. There are too many `private*` scopes in Ruby ;)

However, ruby 1.9.3 does not support `private_constant`: https://ruby-doc.org/core-1.9.3/Module.html

It seems that 2.0.0 support it: https://ruby-doc.org/core-2.0.0/Module.html

IMO we can drop 1.9.3 support, while 2.0.0 is the default version of macOS Sierra.

**Comment by jneen** [comment]
Looks good to me! Bonus points if we can add it to CI.

**Comment by gfx** [comment]
Right. Now run rubocop on CI e223ed3

**Comment by jneen** [comment]
:+1: good to merge :]

**Comment by jneen** [comment]
errrr, actually, i see these warnings in CI:

```
Warning: unrecognized cop Layout found in /home/travis/build/jneen/rouge/.rubocop.yml
Warning: unrecognized cop Bundler found in /home/travis/build/jneen/rouge/.rubocop.yml
Warning: unrecognized cop Lint/EmptyWhen found in /home/travis/build/jneen/rouge/.rubocop.yml
Warning: unrecognized cop Lint/UnneededSplatExpansion found in /home/travis/build/jneen/rouge/.rubocop.yml
```

**Comment by gfx** [comment]
@jneen that's because rubocop is too old on Ruby 1.9.3.

https://github.com/bbatsov/rubocop/blob/master/rubocop.gemspec#L11 the latest rubocop requires ruby >= 2.0.0.

rubocop should be skipped on ruby < 2.0.0.

**Comment by jneen** [comment]
Hm. When I run it locally I see:

```
Warning: unrecognized cop Layout found in /home/jneen/src/rouge/.rubocop.yml
```

**Comment by gfx** [comment]
@jneen Oh, what's your rubocop version?

**Comment by jneen** [comment]
0.47.1 (???)

**Comment by gfx** [comment]
The latest one is v0.49.1 ;)

Speified the version if RUBY_VERSION >= "2.0.0" cee2518

**Comment by gfx** [comment]
Looks good on CI:
screen shot 2017-06-13 at 10 57 45

**Comment by jneen** [comment]
:+1: thanks!

**Comment by gfx** [comment]
💃

**Pull request merged**